### PR TITLE
[6.0] [Completion] Update type attribute completions

### PR DESCRIPTION
**Pull request by hamishknight**
*6.0 cherry-pick of #74890*

- Explanation: Updates type attribute completion to support `@isolated(any)`, `@noDerivative`, `@preconcurrency`, `@retroactive`, `@Sendable`, and `@unchecked`
- Scope: Affects code completion
- Issue: rdar://130741006, rdar://130288443
- Risk: Fairly low, should only affect code completion
- Testing: Added tests to test suite
- Reviewer: Alex Hoppen

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

File: include/swift/AST/TypeAttr.def [hunk 1]
```diff
@@ -83,6 +82,7 @@ SIMPLE_SIL_TYPE_ATTR(inout, Inout)
 SIMPLE_SIL_TYPE_ATTR(inout_aliasable, InoutAliasable)
 SIMPLE_SIL_TYPE_ATTR(in_guaranteed, InGuaranteed)
 SIMPLE_SIL_TYPE_ATTR(in_constant, InConstant)
+SIMPLE_SIL_TYPE_ATTR(noescape, NoEscape)
```
Comment (bnbarham) [inline include/swift/AST/TypeAttr.def:85]: @rjmccall I don't suppose you remember why we didn't make `noescape` (and `async`) a `SIMPLE_SIL_TYPE_ATTR` in https://github.com/swiftlang/swift/commit/41cdfb04adfe13839af9e9867f775559ee307296? Mostly asking in terms of risk here - we could instead leave this change out of 6.0 and just add the missing completions in.

**Review by bnbarham — APPROVED** [review]

File: include/swift/AST/TypeAttr.def (hunk 1 above)
Comment (rjmccall) [reply include/swift/AST/TypeAttr.def:85]: I intentionally didn't put any effort in that patch into recategorizing the existing attributes.  I wanted it to be focused on the high-level representation change.

I don't know why they're not simple attributes in SIL.  They certainly look like simple attributes.

Comment (bnbarham) [reply include/swift/AST/TypeAttr.def:85]: All good, was just checking that there wasn't any underlying reason that we missed. Thanks!

**Pull request merged**